src/linter.cc

```diff
@@ -240,6 +240,59 @@ absl::Status CheckAliasKeyword(absl::string_view sql) {
     }).ApplyTo(sql);
 }
 
+absl::Status CheckTabCharactersUniform(absl::string_view sql,
+    bool allow_all_tabs, const char delimeter) {
+    const char tab = '\t', space = ' ';
+
+    // Find all tabs and spaces in indents.
+    int line_number = 1, line_size = 0;
+    bool is_indent = true;
+    std::vector<CodePosition> tabs, spaces;
+    for (int i=0; i<static_cast<int>(sql.size()); ++i) {
+        if ( sql[i] == delimeter ) {
+            ++line_number;
+            line_size = 0;
+            is_indent = true;
+            continue;
+        } else if ( sql[i] == space ) {
+            if ( is_indent ) {
+                spaces.push_back({ line_number, line_size });
```

Comment:
Why do you save all spaces and tabs? I think just a first occurrence of each is enough, no?